config/config-builder/src/swarm_config.rs

```diff
@@ -175,12 +176,15 @@ impl SwarmConfig {
             // Remove network private keys for this peer.
             let peer_id = PeerId::from_str(&node_id).unwrap();
             warn!("Looking for peer id for peer: {}", node_id);
+            // TODO: Reading the code to figure out why remove_entry below is never going to fail
+            // is tedious and the conclusion that it won't fail is tentative, fragile and subject to change.
+            // Find a better way to be sure that this won't fail.
```

Comment:
1. if this fails, this would fail directly at start up
2. this is not code running in prod, swarm is only use to test things

I think it's worth removing this TODO.

Reply:
Testing code needs to be carefully written as well. If it fails in non-obvious ways it can consume a serious amount of developer time.